src/storage/traits.rs

```diff
@@ -38,25 +40,23 @@ pub struct Component {
 /// so that in case it fails it can be switched over.
 /// We do rely on there being a basic API that can implement this trait though.
 pub trait Backend {
-    /// Get a list of versions for a component
-    fn get_versions(&self, name: &str, loc: Option<&str>) -> LalResult<Vec<u32>>;
+    /// Get a list of versions for a component in descending order
+    fn get_versions(&self, name: &str, loc: &str) -> LalResult<Vec<u32>>;
     /// Get the latest version of a component
-    fn get_latest_version(&self, name: &str, loc: Option<&str>) -> LalResult<u32>;
+    fn get_latest_version(&self, name: &str, loc: &str) -> LalResult<u32>;
 
-    /// Get the tarball url of a `Component` in a backend location
-    /// If no version is given, return latest
-    fn get_tarball_url(
-        &self,
-        name: &str,
-        version: Option<u32>,
-        loc: Option<&str>,
-    ) -> LalResult<Component>;
+    /// Get the version and location information of a component
+    ///
+    /// If no version is given, figure out what latest is
+    fn get_component_info(&self, name: &str, ver: Option<u32>, loc: &str) -> LalResult<Component>;
 
     /// Publish a release build (ARTIFACT dir) to a specific location
```

Comment:
`/// Publish a release build's ARTIFACT to a backend specific location.`
If we're renaming the method, maybe just `publish_artifact`, with a note that this means everything in the directory, the tarball and the lockfile (and maybe a signature when we eventually support those).